### ISSUE TITLE
NVDA and role="presentation" behavior

### ISSUE THREAD

**Issue by noobards** [nvaccess/nvda#5958]
Hi,

I'm very new to the world of accessibility. So excuse me if I'm asking something way too basic. 

I'm using a datepicker and it works perfectly with NVDA. However, there is one small issue which I intend to solve. At present, while navigating between dates, the screen reader announces

> Friday, May 12, 2106 row3
> 
> Monday, April 22, 2016 column4

Now this is because the screen reader is treating the table as data table and hence reading out the extra information about cell co-ordinates. I would like to have it treat the table as a layout table so that it doesn't read out _row3_ or _column4_

But using a _role="presentation"_ on the table tag isn't solving the problem. Is there way I can perhaps programmatically (via javascript) make NVDA NOT read out the cell-ordinate?

PS: I'm aware of the option NVDA has - _Preferences-->Document Formatting-->Uncheck "Report table cell co-ordinates"_ . But can something be done via javascript so that I won't have to depend on the client user disabling it himself?

**NOTE**: Also, can it be confirmed if the datepicker will be USDOT-508 compliant (WCAG 2.1.1) even if the cell co-ordinates are announced by the datepicker?

Here is a demo for reference - [**DEMO**](https://jsfiddle.net/j49gz3g3/4/)

**Comment by poorgeek** [comment]
I've run across this same issue when using NVDA 2016.3 and IE 11. I've put together another [example on CodePen](http://s.codepen.io/poorgeek/debug/ENoQer) that highlights the problem. When you tab from one field to the next, the label is read followed by the row/column information. 

In Firefox, the example reads as expected; ignoring the table semantics.

**Comment by Adriani90** [comment]
cc: @jcsteh, @MarcoZehe your thoughts here would be highly appreciated. @noobards as far as I know at least, there is no possibility to do that. Note that NVDA will read rows and columns coordinates in every layout table if such a table is identified. In NVDA's settings under browse mode settings you can enable consideration of layout tables. After that, NVDA will report row and column coordinates in those tables. This is the behavior on every website containing layout tables. But why do you want the user not to influence this behavior? I think most users would just be fine with turning off cell coordinates in a specific case if they feel more confortable with that.

**Comment by jcsteh** [comment]
role="presentation" is negated if the element is focusable or has aria attributes. This has both (aria-readonly, aria-activedescendant and tabindex). If you want role="presentation" to work, you'd need to remove these. That means the cells themselves would need to accept focus.

**Comment by Adriani90** [comment]
@jcsteh thank you very much for your answer. Since the implementation depends on the web developer and the question of the initial author is now answered, I will close this issue. @noobards if you still have questions, please feel free to open new issues or to comment on this one. Thank you!
